### PR TITLE
[stable/pachyderm] Added ability to deploy the pachyderm dashboard

### PR DESCRIPTION
**Pull request by LaurentGoderre**
Signed-off-by: Laurent Goderre <laurent.goderre@gmail.com>

#### What this PR does / why we need it:
Allows deploying the pachyderm dashboard

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/pachyderm/templates/dash_deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: dash
+  labels:
+    app: {{ template "dash.fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+    suite: {{ template "fullname" . }}
```
Comment (Aisuko) [inline stable/pachyderm/templates/dash_deployment.yaml:11]: Should we add `annotations` here?

File: stable/pachyderm/Chart.yaml [hunk 2]
```diff
@@ -10,7 +10,7 @@ keywords:
   - reproducibility
   - distributed
   - processing
-version: 0.2.1
+version: 0.3.1
```
Comment (Aisuko) [inline stable/pachyderm/Chart.yaml:13]: `0.3.0` could make sense.

File: stable/pachyderm/templates/dash_deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+          resources:
+            requests:
+              cpu: '{{ .Values.dash.proxy.resources.requests.cpu }}'
+              memory: {{ .Values.dash.proxy.resources.requests.memory }}
+            {{- if .Values.dash.proxy.resources.limits }}
+            limits:
+              cpu: '{{ .Values.dash.proxy.resources.limits.cpu }}'
+              memory: {{ .Values.dash.proxy.resources.limits.memory }}
+            {{- end }}
+{{- end }}
```
Comment (Aisuko) [inline stable/pachyderm/templates/dash_deployment.yaml]: Should we add the blank row at the end, that could help to across the CI test.

File: stable/pachyderm/templates/dash_ingress.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  tls:
+  - hosts:
+    {{- if  .Values.dash.ingress.hostName }}
+    - {{ .Values.dash.ingress.hostName }}
+    secretName: {{  .Values.dash.ingress.tlsSecret }}
+    {{- else}}
+    - secretName: {{  .Values.dash.ingress.tlsSecret }}
+    {{- end }}
+{{- end }}
+{{- end }}
```
Comment (Aisuko) [inline stable/pachyderm/templates/dash_ingress.yaml]: Same as above.

File: stable/pachyderm/templates/dash_svc.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    - name: grpc-proxy-http
+      port: {{ .Values.dash.service.port.grpc }}
+      targetPort: 8081
+      {{- if eq .Values.dash.service.type "NodePort" }}
+      nodePort: {{ .Values.dash.service.nodePort.grpc }}
+      {{- end }}
+  selector:
+    app: {{ template "dash.fullname" . }}
+  type: {{ .Values.dash.service.type | quote }}
+{{- end }}
```
Comment (Aisuko) [inline stable/pachyderm/templates/dash_svc.yaml]: Same as above.

File: stable/pachyderm/templates/pachd_svc.yaml [hunk 6]
```diff
@@ -2,67 +2,38 @@
 kind: Service
 apiVersion: v1
 metadata:
-  name: pachd-api-grpc
-  {{- if .Values.pachd.service.grpc.annotations }}
+  name: pachd
+  {{- if .Values.pachd.service.annotations }}
   annotations:
-    {{ .Values.pachd.service.grpc.annotations | toYaml | indent 4 | trim  }}
+    {{ .Values.pachd.service.annotations | toYaml | indent 4 | trim  }}
```
Comment (Aisuko) [inline stable/pachyderm/templates/pachd_svc.yaml:8]: `{{ toYaml .Values.pachd.service.annotations | indent 4 }}` could be make sense right?

File: stable/pachyderm/values.yaml [hunk 7]
```diff
@@ -98,3 +104,56 @@ etcd:
...
+  ## Set to true to enable ingress record generation
+  enabled: false
+  ## hostName: dash
+  path: /
+  ## Set this to true in order to enable TLS on the ingress record
+  tls: false
+  ## If TLS is set to true, you must declare what secret will store the key/certificate for TLS
+  tlsSecret: myTlsSecret
+  ## Ingress annotations done as key:value pairs
+  annotations:
```
Comment (Aisuko) [inline stable/pachyderm/values.yaml]: `annotations: {}` could be make sense right?

**Review by Aisuko — CHANGES_REQUESTED** [review]
Thanks for your PR, there may some piece of advice. Oh, please signoff DCO

**Comment by jonandernovella** [comment]
Looks good to me. Those minor changes should do it @LaurentGoderre

File: stable/pachyderm/templates/dash_deployment.yaml (hunk 1 above)
Comment (LaurentGoderre) [reply stable/pachyderm/templates/dash_deployment.yaml:11]: Neither the `pachd` or `etcd` deployments have annotations there so I wouldn't add it in this PR.

**Review by jonandernovella — APPROVED** [review]
LGTM!

**Comment by LaurentGoderre** [comment]
It isn't quire ready, it deploys but the UI doesn't work because of the web sockets.

**Review by jonandernovella — CHANGES_REQUESTED** [review]
We first need to fix the web sockets issue before merging this PR.

**Comment by zanhsieh** [comment]
up

**Comment by jonandernovella** [comment]
@LaurentGoderre Your second commit needs signoff before we can marge it.

**Pull request closed without merging**